datafusion/core/src/datasource/physical_plan/avro.rs

```diff
@@ -18,24 +18,31 @@
 //! Execution plan for reading line-delimited Avro files
 
 use std::any::Any;
+use std::fmt::Formatter;
 use std::sync::Arc;
 
-use super::FileScanConfig;
+use super::{FileOpener, FileScanConfig};
+#[cfg(feature = "avro")]
+use crate::datasource::avro_to_arrow::Reader as AvroReader;
+use crate::datasource::data_source::{FileSource, FileType};
 use crate::error::Result;
-use crate::physical_plan::metrics::{ExecutionPlanMetricsSet, MetricsSet};
-use crate::physical_plan::{
-    DisplayAs, DisplayFormatType, ExecutionPlan, Partitioning, PlanProperties,
-    SendableRecordBatchStream, Statistics,
-};
 
 use arrow::datatypes::SchemaRef;
-use datafusion_common::Constraints;
-use datafusion_execution::TaskContext;
-use datafusion_physical_expr::{EquivalenceProperties, LexOrdering};
+use datafusion_common::{Constraints, Statistics};
+use datafusion_execution::{SendableRecordBatchStream, TaskContext};
+use datafusion_physical_expr::{EquivalenceProperties, Partitioning};
+use datafusion_physical_expr_common::sort_expr::LexOrdering;
 use datafusion_physical_plan::execution_plan::{Boundedness, EmissionType};
+use datafusion_physical_plan::metrics::{ExecutionPlanMetricsSet, MetricsSet};
+use datafusion_physical_plan::{
+    DisplayAs, DisplayFormatType, ExecutionPlan, PlanProperties,
+};
+
+use object_store::ObjectStore;
 
 /// Execution plan for scanning Avro data source
 #[derive(Debug, Clone)]
+#[deprecated(since = "46.0.0", note = "use DataSourceExec instead")]
```

Comment:
Thank you for deprecating this.

In order to avoid having to maintain 2 copies of the code (and avoid having them drift out of sync), I think we should consider leaving the interface the same but changing the internal implementation. 

So that would be something like:
```rust
pub struct AvroExec {
    inner: DataSourceExec, // wraps a DataSourceExec
}
```